frontend/packages/dev-console/src/components/catalog/hooks/useCatalogExtensions.ts

```diff
@@ -0,0 +1,29 @@
+import {
+  CatalogItemProvider,
+  CatalogItemType,
+  isCatalogItemProvider,
+  isCatalogItemType,
+} from '@console/plugin-sdk';
+import {
+  ResolvedExtension,
+  useResolvedExtensions,
+} from '@console/dynamic-plugin-sdk/src/api/useResolvedExtensions';
+
+const useCatalogExtensions = (
+  catalogType: string,
+): [ResolvedExtension<CatalogItemType>[], ResolvedExtension<CatalogItemProvider>[]] => {
+  const itemTypeExtensions = useResolvedExtensions<CatalogItemType>(isCatalogItemType);
+  const itemProviderExtensions = useResolvedExtensions<CatalogItemProvider>(isCatalogItemProvider);
```

Comment:
You mean have ability to in useResolvedExtensions to resolve specific types of extensions?

Reply:
We don't need to do this now but I was just suggesting something like this to prevent resolving of references we don't yet intend on using:
```suggestion
   const itemTypeExtensions = useResolvedExtensions<CatalogItemType>(isCatalogItemType);
   const itemProviderExtensions = useResolvedExtensions<CatalogItemType>(isCatalogItemProvider);
  const resolvedItemTypeExtensions = useResolvedExtensions(itemTypeExtensions.filter(e => e.properties.type === catalogType);
  const resolvedItemProviderExtensions = useResolvedExtensions(itemProviderExtensions.filter(e => e.properties.type === catalogType);
```